### PR TITLE
Mark credentials.oauthClient.token_endpoint_auth_method as nullable for oauth apps.

### PR DESCRIPTION
**Pull request by danielzeller-okta**
## Description:
- All elements of the `credentials.oauthClient` object on the /apps API are nullable
- Mark `token_endpoint_auth_method` as such

### Resolves:
* [OKTA-179275](https://oktainc.atlassian.net/browse/OKTA-179275)

## Primary Reviewer:
@yuliu-okta @jakubvul 

### Additional Reviewers
@federations-okta

### CODE REVIEW

**Comment by jakubvul** [comment]
FYI for @eklein-okta this is not officially related to a release...yet.

File: _source/_docs/api/resources/apps.md [hunk 1]
```diff
@@ -990,7 +990,7 @@ Adds an OAuth 2.0 client application. This application is only available to the
 |:---------------------------|:---------------------------------------------------------------|:----------------------------------------------------------------------------|:---------|:-------|:-----------|
 | client_id                  | Unique identifier for the client application                   | String                                                                      | TRUE     | TRUE   | TRUE       |
 | client_secret              | OAuth 2.0 client secret string (used for confidential clients) | String                                                                      | TRUE     | FALSE  | TRUE       |
-| token_endpoint_auth_method | Requested authentication method for the token endpoint         | `none`, `client_secret_post`, `client_secret_basic`, or `client_secret_jwt` | FALSE    | FALSE  | FALSE      |
+| token_endpoint_auth_method | Requested authentication method for the token endpoint         | `none`, `client_secret_post`, `client_secret_basic`, or `client_secret_jwt` | TRUE     | FALSE  | FALSE      |
```
Comment (yuliu-okta) [inline _source/_docs/api/resources/apps.md]: It is better to mention the default value is `client_secret_basic`

**Comment by danielzeller-okta** [comment]
After consultation with @jakubvul, I've just gone ahead and removed the Nullable, Unique and Validation columns entirely, as this is something that's going to happen in the future anyways.

image

File: _source/_docs/api/resources/apps.md [hunk 2]
```diff
@@ -1003,7 +1003,7 @@ Adds an OAuth 2.0 client application. This application is only available to the
 | redirect_uris                             | Array of redirection URI strings for use in redirect-based flows                            | Array                                                                                        | TRUE     | FALSE  | TRUE       |           |
 | response_types                            | Array of OAuth 2.0 response type strings                                                    | Array of `code`, `token`, `id_token`                                                         | TRUE     | FALSE  | TRUE       |           |
 | grant_types                               | Array of OAuth 2.0 grant type strings                                                       | Array of `authorization_code`, `implicit`, `password`, `refresh_token`, `client_credentials` | FALSE    | FALSE  | TRUE       |           |
-| initiate_login_uri                        | URL string that a third party can use to initiate a login by the client                     | String                                                                                       | TRUE     | FALSE  | TRUE       |           |
```
Comment (yuliu-okta) [inline _source/_docs/api/resources/apps.md]: why change this line?

Comment (jakubvul) [reply _source/_docs/api/resources/apps.md]: Actually, the "Default" column in this table is almost entirely empty and therefore useless. Can we use this as an opportunity to just delete it entirely? The two params that do have entries in that column can have that information moved to the Description.

Comment (danielzeller-okta) [reply _source/_docs/api/resources/apps.md]: Removed the Default column

**Comment by danielzeller-okta** [comment]
After removing "Default" column:
image

File: _source/_docs/api/resources/apps.md [hunk 3]
```diff
@@ -986,29 +986,29 @@ Adds an OAuth 2.0 client application. This application is only available to the
...
+| Parameter                  | Description                                                                                                                                       | DataType | Default               |
+|:---------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------|:---------|:----------------------|
+| client_id                  | Unique identifier for the client application                                                                                                      | String   |                       |
+| client_secret              | OAuth 2.0 client secret string (used for confidential clients)                                                                                    | String   |                       |
+| token_endpoint_auth_method | Requested authentication method for the token endpoint. Valid values: `none`, `client_secret_post`, `client_secret_basic`, or `client_secret_jwt` | String   | `client_secret_basic` |
```
Comment (yuliu-okta) [inline _source/_docs/api/resources/apps.md]: It looks like in other places, the enum value is listed in the DataType instead of in the description.

Comment (jakubvul) [reply _source/_docs/api/resources/apps.md]: Yes, and this is a bad thing. We should not reproduce previous mistakes.

Comment (danielzeller-okta) [reply _source/_docs/api/resources/apps.md]: Yes it is, but I chatted with @jakubvul about this and he preferred that I put it in Description, and have the DataType be String.

**Comment by danielzeller-okta** [comment]
@eklein-okta This is now in the 2018.29 release

**Comment by nbarbettini** [comment]
Merging for 2018.29 release (see #2200).

**Pull request merged**